zul/src/archive/web/js/zul/grid/Row.js

```diff
@@ -174,12 +174,18 @@ zul.grid.Row = zk.$extends(zul.Widget, {
 		var out = opts.out || new zk.Buffer(),
 			child = opts.child,
 			isCell = child.$instanceof(zul.wgt.Cell);
+
+		if (isCell)
+			child._headerVisible = opts.visible;
+
 		if (!isCell) {
 			out.push('<td id="', child.uuid, '-chdextr"',
 				this._childAttrs(child, opts.index, !opts.visible), '><div id="', child.uuid,
 				'-cell" class="', opts.zclass, '-content">');
 		}
```

Comment:
opts.visible value pass into this._childAttrs, could use it instead of new attr ? (child._headerVisible)